web/src/hotspots.ts

```diff
@@ -103,11 +121,14 @@ function compute_placement($elt, popover_height, popover_width, prefer_vertical_
     return placement;
 }
 
-export function post_hotspot_as_read(hotspot_name) {
+export function post_hotspot_as_read(hotspot_name: string): void {
     onboarding_steps.post_onboarding_step_as_read(hotspot_name);
 }
 
-function place_icon(hotspot) {
+function place_icon(hotspot: Hotspot): boolean {
+    if (hotspot.location === undefined) {
+        return false;
+    }
```

Comment:
This one should be an assert too. 

Or actually probably the bug is that `hotspot_schema` should be split into a `RawHotspot` that's the server type, and a `Hotspot` that guarantees the `.location` field exists, and this check would just not be required.

Reply:
Splitting it will definitely avoid issues at multiple instances. Will move forward in that direction.